cabal-testsuite/PackageTests/NewBuild/Semaphore/cabal.test.hs

```diff
@@ -0,0 +1,6 @@
+import Test.Cabal.Prelude
+main = cabalTest $ do
+    skipUnlessGhcVersion ">= 9.7"
+    -- Parallel flag means output of this test is nondeterministic
+    recordMode DoNotRecord $
+        cabal "v2-build" ["-j", "--semaphore", "Semaphore"]
```

Comment:
So semaphore is not boolean, but also accepts a name?

Reply:
Looking at this screenshot, it would appear that cabal automatically translates the --semaphore option to --semaphore=unique_semaphore_name 

![Screenshot_20230714_110247](https://github.com/haskell/cabal/assets/29253044/bc5db975-b635-46ed-8630-28cb5d649a5a)